### PR TITLE
Support nullable references.

### PR DESCRIPTION
**Pull request by ejball**

### CODE REVIEW

File: src/Faithlife.Data/DbConnectorCommand.cs [hunk 1]
```diff
@@ -302,7 +302,7 @@ private IReadOnlyList<T> DoQuery<T>(Func<IDataRecord, T> read)
 			return list;
 		}
 
-		private T DoQueryFirst<T>(Func<IDataRecord, T> read, bool single, bool orDefault)
+		private T DoQueryFirst<T>(Func<IDataRecord, T>? read, bool single, bool orDefault)
```
Comment (bgrainger) [inline src/Faithlife.Data/DbConnectorCommand.cs]: Need to add `[return: MaybeNull]`?

File: src/Faithlife.Data/DbValueTypeInfo.cs [hunk 2]
```diff
@@ -104,7 +104,7 @@ public T GetValue(IDataRecord record, int index, int count)
 			else if (m_strategy == DbValueTypeStrategy.ByteArray)
 			{
 				if (record.IsDBNull(index))
-					return default;
+					return default!;
```
Comment (bgrainger) [inline src/Faithlife.Data/DbValueTypeInfo.cs]: Can `!` be avoided with `[return: MaybeNull]`?

File: src/Faithlife.Data/DbConnectorCommand.cs (hunk 1 above)
Comment (ejball) [reply src/Faithlife.Data/DbConnectorCommand.cs]: I toyed with that but it wasn't clear what good it was doing. I'd rather not add it until I understand why it is useful.

File: src/Faithlife.Data/DbValueTypeInfo.cs (hunk 2 above)
Comment (ejball) [reply src/Faithlife.Data/DbValueTypeInfo.cs]: Not IIRC, but I'll double-check.

Comment (ejball) [reply src/Faithlife.Data/DbValueTypeInfo.cs]: Confirmed. Doesn't seem to help.

File: src/Faithlife.Data/DbConnectorCommand.cs (hunk 1 above)
Comment (bgrainger) [reply src/Faithlife.Data/DbConnectorCommand.cs]: You're coercing `default(T)` with `!` in this method, so it's pretty clear the method can return `null`. The `MaybeNull` attribute is designed for this exact use case.

I'd rather you not add nullable annotations until you understand why the attributes are needed.

File: src/Faithlife.Data/DbValueTypeInfo.cs (hunk 2 above)
Comment (bgrainger) [reply src/Faithlife.Data/DbValueTypeInfo.cs]: Hmm, I'm surprised, but I'd have to check it out in a small repro.

File: src/Faithlife.Data/DbConnectorCommand.cs (hunk 1 above)
Comment (ejball) [reply src/Faithlife.Data/DbConnectorCommand.cs]: I understand the fundamental principle behind the attribute, but documentation is sparse, the attribute wasn't having the effect I expected (see below), and I can't use it when building .NET Standard 2.0, so I let it go. Happy for any assistance you can give.

Comment (ejball) [reply src/Faithlife.Data/DbConnectorCommand.cs]: da8c6694249ce9936550a595119d4c2e93407a99

**Pull request merged**